stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1897,3 +1897,46 @@ end
 
 normalize(x) = x / norm(x)
 normalize(x, p::Real) = x / norm(x, p)
+
+"""
+    copytotri!(uplo, A, B) -> B
+
+Copies a triangular part of a matrix `A` to another matrix `B`.
+`uplo` specifies the part of the matrix `A` to be copied to `B`.
+Set `uplo = 'L'` for the lower triangular part or `uplo = 'U'
+for the upper triangular part.
+
+!!! compat "Julia 1.11"
+    `copytotri!` requires at least Julia 1.11.
+
+# Examples
+```jldoctest
+julia> A = [1 2 ; 3 4];
+
+julia> B = [0 0 ; 0 0];
+
+julia> copytotri!('L', A, B)
+2×2 Matrix{Int64}:
+ 1  0
+ 3  4
+```
+"""
+function copytotri!(uplo::AbstractChar, A::AbstractMatrix, B::AbstractMatrix)
+    require_one_based_indexing(A, B)
+    LinearAlgebra.BLAS.chkuplo(uplo)
+    m, n = size(A)
```

Comment:
You definitely need to check the size of `B` as well, otherwise (if `B` is smaller than `A`) some indices might be out-of-bounds. Judging by the LAPACK version below, you do want to allow that `B` is larger than `A`, right?

Reply:
Yes, we should allow `B` to be larger than `A`.

>